agent/taskresource/credentialspec/credentialspec_windows_test.go

```diff
@@ -80,18 +86,22 @@ func TestClearCredentialSpecDataErr(t *testing.T) {
 
 	credSpecMapData := map[string]string{
 		"credentialspec:file://localfilePath.json": "credentialspec=file://localfilePath.json",
-		"credentialspec:s3ARN":                     "credentialspec=file://ResourceDir/s3_taskARN_fileName.json",
-		"credentialspec:ssmARN":                    "credentialspec=file://ResourceDir/ssm_taskARN_fileName.json",
+		"credentialspec:s3ARN":                     "credentialspec=file://s3_taskARN_fileName.json",
+		"credentialspec:ssmARN":                    "credentialspec=file://ssm_taskARN_fileName.json",
 	}
 
+	credentialSpecResourceLocation := "C:/ProgramData/docker/credentialspecs/"
 	credspecRes := &CredentialSpecResource{
-		CredSpecMap: credSpecMapData,
-		os:          mockOS,
+		CredSpecMap:                    credSpecMapData,
+		os:                             mockOS,
+		credentialSpecResourceLocation: credentialSpecResourceLocation,
 	}
 
+	expectedS3FilePath := filepath.Join(credentialSpecResourceLocation, "s3_taskARN_fileName.json")
+	expectedSSMFilePath := filepath.Join(credentialSpecResourceLocation, "ssm_taskARN_fileName.json")
 	gomock.InOrder(
-		mockOS.EXPECT().Remove(gomock.Any()).Return(nil).Times(1),
-		mockOS.EXPECT().Remove(gomock.Any()).Return(errors.New("test error")),
+		mockOS.EXPECT().Remove(expectedS3FilePath).Return(nil),
```

Comment:
This relies on the map iteration order the same as the order you saved in the map. I don't think this is guaranteed to the truth based on how hashmap is implemented. This will introduce the flakiness to the test. This does not need to be in a specific order.

Reply:
Done, thanks!